zlib-rs/src/deflate.rs

```diff
@@ -974,10 +975,21 @@ impl<'a> BitWriter<'a> {
         self.send_bits_trace(val, len);
         self.sent_bits_add(len as usize);
 
-        self.bit_buffer |= val << self.bits_used;
         if total_bits < Self::BIT_BUF_SIZE {
+            self.bit_buffer |= val << self.bits_used;
             self.bits_used = total_bits;
         } else {
+            self.send_bits_overflow(val, total_bits);
+        }
+    }
+
+    fn send_bits_overflow(&mut self, val: u64, total_bits: u8) {
```

Comment:
you can try if adding `#[cold]` here makes a difference. It doesn't really for me. Otherwise this looks good!